### PR TITLE
media-tv/v4l-dvb-saa716x: fix bugs, remove obsolete

### PR DESCRIPTION
**Pull request by madmartin**
make missing kernel configurations in CONFIG_CHECK fatal
adapt BUILD_PARAMS for upcoming kernel 5.3 and up
no version bump for current ebuild
remove obsolete ebuild 0.0.1_p20170225-r1

Closes: https://bugs.gentoo.org/690372
Closes: https://bugs.gentoo.org/696292
Package-Manager: Portage-2.3.79, Repoman-2.3.17
Signed-off-by: Martin Dummer <martin.dummer@gmx.net>

### CODE REVIEW

File: media-tv/v4l-dvb-saa716x/v4l-dvb-saa716x-0.0.1_p20170225-r2.ebuild [hunk 1]
```diff
@@ -52,9 +53,12 @@ src_prepare() {
...
-	BUILD_PARAMS="SUBDIRS=${S}/linux/drivers/media/common/saa716x \
-	CONFIG_SAA716X_CORE=m CONFIG_DVB_SAA716X_FF=m CONFIG_DVB_SAA716X_BUDGET=m \
-	CONFIG_DVB_SAA716X_HYBRID=m"
-	addpredict "${EROOT%/}"/usr/src/linux/
+	kernel_is le 5 2 && BUILD_PARAMS="SUBDIRS=${S}/linux/drivers/media/common/saa716x \
+		CONFIG_SAA716X_CORE=m CONFIG_DVB_SAA716X_FF=m CONFIG_DVB_SAA716X_BUDGET=m \
+		CONFIG_DVB_SAA716X_HYBRID=m"
+	kernel_is gt 5 2 && BUILD_PARAMS="M=${S}/linux/drivers/media/common/saa716x/ \
+		CONFIG_SAA716X_CORE=m CONFIG_DVB_SAA716X_FF=m CONFIG_DVB_SAA716X_BUDGET=m \
+		CONFIG_DVB_SAA716X_HYBRID=m"
```
Comment (cjmayo) [inline media-tv/v4l-dvb-saa716x/v4l-dvb-saa716x-0.0.1_p20170225-r2.ebuild]: Could this be a bit more concise - I don't see any other differences between the BUILD_PARAMS e.g.:

```sh
	kernel_is le 5 2 && BUILD_PARAMS="SUBDIRS" || BUILD_PARAMS="M"
	BUILD_PARAMS+="=${S}/drivers/media/pci/saa716x \
```

Comment (madmartin) [reply media-tv/v4l-dvb-saa716x/v4l-dvb-saa716x-0.0.1_p20170225-r2.ebuild]: Yes, good idea.
I tested a little an figured out that `BUILD_PARAMS="M"` works from kernel 5.1 up - will change that, too.

**Comment by juippis** [comment]
Hmm, I don't agree with these kernel checks being fatal. Imagine running `-uavDN world` with 100 packages, which then stops because you need to update your kernel before you can continue, due to some "module" package. 

Well ultimately it's up for you to decide, but just my 2cents. systemd and networkmanager require certain kernel options to run them, but they are not fatal to my knowledge either. It's easier to continue after updates have finished.

You can close / I can close the bugs as WONTFIX to resolve them.

**Comment by madmartin** [comment]
> Hmm, I don't agree with these kernel checks being fatal.

Turning the kernel config checks to fatal is a decision I made after reading **linux-info.eclass**:

```
# In general, most checks should be non-fatal. The only time fatal checks should
# be used is for building kernel modules or cases that a compile will fail
# without the option.
#
# This is to allow usage of binary kernels, and minimal systems without kernel
# sources.
```
I see both indications true here - we DO build kernel modules, and if the necessary config values in the kernel are not present, the module compile WILL definitely fail. Both referenced bugs have this cause - the kernel used for ebuild testing does not have the prerequisites set.

> Imagine running `-uavDN world` with 100 packages, which then stops because you need to update your kernel before you can continue, due to some "module" package.

Yes, I imagine, and add  `--keep-going` to the emerge commandline. So IMHO this example does not count. Any of the 100 packages can break the chain.

> 
> Well ultimately it's up for you to decide, but just my 2cents. ...
> You can close / I can close the bugs as WONTFIX to resolve them.

I would follow guidance of  **linux-info.eclass** an keep CONFIG_CHECK fatal.

**Comment by juippis** [comment]
Your sign-off was malformed btw. It seems like you use repoman to commit, so it should sign them correctly automatically? What happened?

**Comment by madmartin** [comment]
@juippis I dont know why the < > got removed, I did not notice that. 
Usually, after fixing a commit, I execute 
`git rebase --exec 'git commit --amend --no-edit -n -sS' -i <commit-before-mine>`
to add sign-off and gpg signature on every single commit.

**Pull request closed without merging**